### PR TITLE
elasticsearch: walker for query generation

### PR DESCRIPTION
**Pull request by jirikuncar**
* NEW Adds **experimental** support for Elasticsearch query generation.

Signed-off-by: Jiri Kuncar <jiri.kuncar@cern.ch>

### CODE REVIEW

**Comment by jirikuncar** [comment]
@SamiHiltunen can you have a look to old tests (https://github.com/inveniosoftware/invenio/blob/maint-2.1/invenio/modules/search/testsuite/test_elasticsearch_walker.py) and try to port some more?

File: .travis.yml [hunk 1]
```diff
@@ -24,7 +24,6 @@ language: python
 sudo: false
 
 python:
-  - "2.6"
```
Comment (tiborsimko) [inline .travis.yml]: :+1: for killing Python 2.6

File: invenio_query_parser/contrib/elasticsearch/walkers/dsl.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                ]}}
+            return {'term': {field: node.value for field in fields}}
+        return query
+
+    @visitor(RegexValue)
+    def visit(self, node):
+        def query(keyword):
+            fields = self.get_fields_for_keyword(keyword, mode='r')
+            if len(fields) > 1:
+                res = {"bool": {"should": []}}
```
Comment (jirikuncar) [inline invenio_query_parser/contrib/elasticsearch/walkers/dsl.py]: note to myself: refactor like in https://github.com/inveniosoftware/invenio-query-parser/pull/41/files#diff-2cc2bfe6afba2c08dc65923ec65e9ccfR111

File: invenio_query_parser/contrib/elasticsearch/walkers/dsl.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                res["bool"]["should"] = [{'regexp': {k: node.value}}
+                                         for k in fields]
+            elif fields[0] != "_all":
```
Comment (jirikuncar) [inline invenio_query_parser/contrib/elasticsearch/walkers/dsl.py]: refactor to use `keyword is None` at the beginning

**Pull request merged**